actors/builtin/miner/miner_state.go

```diff
@@ -133,28 +133,29 @@ type WorkerKeyChange struct {
 	EffectiveAt abi.ChainEpoch
 }
 
+// possible states
+// (DealIDs > 0, UnsealedCID != nil) => CommD provided by SP, needs to be checked
+// (DealIDs > 0, UnsealedCID == nil) => Old method or migration, DealIDs checked at PreCommit (Depracated)
+// (DealIDs == 0, UnsealedCID == nil) => NoDeals, CommD of empty sector
+// (DealIDs == 0, UnsealedCID != 0) future approach, fail for now
```

Comment:
Having seen the code now, I would really like to reduce the set of possibilities here by denying the nil unsealed CID. For the CommD of empty sector, we can use a placeholder just to save the bytes. For the old method let's fill it in before persisting this object to state. I'll write more in overall review comment.

Reply:
Issue them becomes how do we handle the migration, do we want to compute CommDs during it?
If not we have to allow for nil CommD.

Reply:
Yes, we can compute them.